src/main/java/org/broadinstitute/hellbender/tools/spark/sv/GatherSplitReadsSpark.java

```diff
@@ -44,20 +49,120 @@ protected void runTool( final JavaSparkContext ctx )
         if ( header.getSortOrder() != SAMFileHeader.SortOrder.coordinate )
             throw new GATKException("The BAM must be coordinate sorted.");
 
-        final JavaRDD<GATKRead> clusteredReads =
+        // Process the input data to produce a map of Kmer -> List of breakpoint IDs.
+        // (A breakpoint ID is just an arbitrary identifier for the putative breakpoint.)
+        // See the doc for getReadPullingMap to see a description of how this works.
+        final Broadcast<Map<Kmer,List<Long>>> broadcastMap = ctx.broadcast(getReadPullingMap(ctx));
```

Comment:
On first reading, it wasn't immediately apparent to me that getReadPullingMap() was actually launching a big Spark pipeline where the bulk of the clustering work is happening. I might make that a little more clear through different method names or commenting.